File: Npc/11001714.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11001714: Tinchai
/// </summary>
public class _11001714 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0728022507006966$
    // - Hello there.
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0805021607007090$
                // - None of us are okay right now. All the more reason to stay strong.
                switch (selection) {
                    // $script:0805021607007091$
                    // - (Nod your head.)
                    case 0:
                        return 40;
                }
                return -1;
            case (40, 0):
                // $script:0805021607007092$
                // - You know, I... Oh, never mind.
                //   <font color="#909090">($npcName:11001714[gender:1]$ looks you in the eye, but stops herself from saying more.)</font>
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.SelectableDistractor,
            (40, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11001714.py
""" 11001714: Tinchai """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0728022507006966$
        # - Hello there.
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0805021607007090$
        # - None of us are okay right now. All the more reason to stay strong.
        if pick == 0:
            # $script:0805021607007091$
            # - (Nod your head.)
            return 40
        return -1

    def __40(self, pick: int) -> int:
        # $script:0805021607007092$
        # - You know, I... Oh, never mind.
        #   <font color="#909090">($npcName:11001714[gender:1]$ looks you in the eye, but stops herself from saying more.)</font>
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.SELECTABLE_DISTRACTOR
        elif (self.state, self.index) == (40, 0):
            return Option.CLOSE
        return Option.NONE
